text/0026-rest-api/rest-api.yaml

```diff
@@ -0,0 +1,983 @@
+openapi: 3.0.3
+info:
+  title: IOTA REST API
+  description: This document describes the upcoming REST API for Hornet and Bee.
+  contact:
+    email: contact@iota.org
+  license:
+    name: Apache 2.0
+    url: 'http://www.apache.org/licenses/LICENSE-2.0.html'
+  version: 1.0.0
+externalDocs:
+  description: Find out more about IOTA
+  url: 'https://iota.org'
+servers:
+  - url: 'http://127.0.0.1:14265'
+tags:
+  - name: node
+    description: Everything about the node itself.
+  - name: tangle
+    description: Everything about the tangle itself.
+  - name: messages
+    description: Everything about messages.
+  - name: UTXO
+    description: Everything about UTXOs.
+  - name: milestones
+    description: Everything about milestones.
+
+paths:
+
+  '/health':
+    get:
+      tags:
+        - node
+      summary: Returns the health of the node.
+      description: Returns the health of the node, which can be used for load-balancing or uptime monitoring.
+      responses:
+        '200':
+          description: node is healthy
+        '503':
+          description: node is not healthy
+
+  '/api/v1/info':
+    get:
+      tags:
+        - node
+      summary: Returns information about the node.
+      description: Returns information about the node.
+      responses:
+        '200':
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/GetInfoResponse'
+              examples:
+                default:
+                  $ref: '#/components/examples/get-info-response-example'
+  
+  '/api/v1/tips':
+    get:
+      tags:
+        - tangle
+      summary: Returns two non-lazy tips.
+      description: >-
+        Returns two non-lazy tips. In case the node can only provide one tip,
+        tip1 and tip2 are identical.
+      responses:
+        '200':
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/GetTipsResponse'
+              examples:
+                Normal:
+                  $ref: '#/components/examples/get-tips-response-example-normal'
+                Identical Tips:
+                  $ref: '#/components/examples/get-tips-response-example-same'
+  
+  '/api/v1/messages':
+    post:
+      tags:
+        - messages
+      summary: Submit a message to the node.
```

Comment:
If it sends only one message, this endpoint should be `/api/v1/message` imho.

Reply:
The GET all and add/append endpoints are normally in plural form and the url to address a single resource is in singular form.

Reply:
Some people like to do all the endpoints in plural form. If you would like that better we can change it.

Reply:
Leave it as is then. I think making other APIs all in plural from will be a little bit confusing. I'm fine with the this as long as description mentions it's one message.